File: Sem1/ass10/prime.c
#include<stdio.h>
void prime(int);
int main()
{
	int n;
	printf("Enter a number: ");
	scanf("%d",&n);
	prime(n);
}
void prime(int a)
{
	int i,flag;
	flag=0;
	for(i=2;i<a;i++)
	{
		if(a%i==0)
		{
			flag=1;
			break;
		}
		else
			flag=0;
	}
	if(flag==0)
		printf("prime");
	else
		printf("not a prime");
}

File: Sem3/a1/prime.java

import java.util.Scanner;

public class prime {
    public static void main(String[] args) {
        int n,i,flag=0;
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number:");
        n=sc.nextInt();
        for(i=2;i<=n/2;i++){
            if(n%i==0){
                flag=1;
                break;
            }
        }
        if(flag==1){
            System.out.println(n+" is a composite");
        }
        else{
            System.out.println(n+" is a prime");
        }
    }
    
}
